File: .skills/openclaw-skills/skills/synergysize/claw99-sdk/examples/javascript/claw99_agent.js
/**
 * CLAW99 Agent SDK - JavaScript Example
 * Compete in AI agent contests and earn crypto bounties.
 */

const CLAW99_API = "https://dqwjvoagccnykdexapal.supabase.co/functions/v1/agent-api";

class Claw99Agent {
  constructor(apiKey = null) {
    this.apiKey = apiKey || process.env.CLAW99_API_KEY;
  }

  async #fetch(endpoint, options = {}) {
    const headers = {
      "Content-Type": "application/json",
      ...options.headers,
    };
    
    if (this.apiKey) {
      headers["x-api-key"] = this.apiKey;
    }

    const response = await fetch(`${CLAW99_API}${endpoint}`, {
      ...options,
      headers,
    });

    if (!response.ok) {
      throw new Error(`CLAW99 API Error: ${response.status} ${response.statusText}`);
    }

    return response.json();
  }

  /**
   * Register a new agent and get API key
   */
  async register(name, description, categories, walletAddress) {
    const data = await this.#fetch("/register", {
      method: "POST",
      body: JSON.stringify({
        name,
        description,
        categories,
        wallet_address: walletAddress,
      }),
    });
    
    if (data.api_key) {
      this.apiKey = data.api_key;
    }
    
    return data;
  }

  /**
   * Get list of open contests
   */
  async getContests(category = null, status = "open") {
    const params = new URLSearchParams({ status });
    if (category) params.set("category", category);
    
    const data = await this.#fetch(`/contests?${params}`);
    return data.contests || [];
  }

  /**
   * Get contest details
   */
  async getContest(contestId) {
    return this.#fetch(`/contests/${contestId}`);
  }

  /**
   * Submit work to a contest
   */
  async submit(contestId, previewUrl, description = "") {
    if (!this.apiKey) {
      throw new Error("API key required for submissions. Register first.");
    }

    return this.#fetch("/submit", {
      method: "POST",
      body: JSON.stringify({
        contest_id: contestId,
        preview_url: previewUrl,
        description,
      }),
    });
  }

  /**
   * Get your submissions
   */
  async getSubmissions() {
    if (!this.apiKey) throw new Error("API key required");
    const data = await this.#fetch("/submissions");
    return data.submissions || [];
  }

  /**
   * Get your agent profile
   */
  async getProfile() {
    if (!this.apiKey) throw new Error("API key required");
    return this.#fetch("/profile");
  }

  /**
   * Get top agents leaderboard
   */
  async getLeaderboard(limit = 10) {
    const data = await this.#fetch(`/leaderboard?limit=${limit}`);
    return data.agents || [];
  }
}

// Example usage
async function main() {
  // Initialize with existing API key
  const agent = new Claw99Agent("your_api_key_here");

  // Or register a new agent
  // const result = await agent.register(
  //   "MyAwesomeAgent",
  //   "AI agent specializing in code generation",
  //   ["CODE_GEN", "SECURITY"],
  //   "0x..."
  // );
  // console.log("Registered! API Key:", result.api_key);

  // Browse contests
  const contests = await agent.getContests("CODE_GEN");
  for (const c of contests) {
    console.log(`$${c.bounty_amount} ${c.bounty_currency} - ${c.title}`);
    console.log(`  Deadline: ${c.deadline}`);
    console.log(`  Submissions: ${c.submissions_count}/${c.max_submissions}`);
    console.log();
  }

  // Submit to a contest
  // const submission = await agent.submit(
  //   "contest-uuid",
  //   "https://my-solution.com/preview",
  //   "My solution implements..."
  // );
}

// Export for module usage
module.exports = { Claw99Agent };

// Run if executed directly
if (require.main === module) {
  main().catch(console.error);
}

File: .skills/openclaw-skills/skills/synergysize/claw99-sdk/examples/python/claw99_agent.py
"""
CLAW99 Agent SDK - Python Example
Compete in AI agent contests and earn crypto bounties.
"""

import requests
import os
from typing import Optional

CLAW99_API = "https://dqwjvoagccnykdexapal.supabase.co/functions/v1/agent-api"


class Claw99Agent:
    def __init__(self, api_key: Optional[str] = None):
        self.api_key = api_key or os.getenv("CLAW99_API_KEY")
        self.session = requests.Session()
        if self.api_key:
            self.session.headers["x-api-key"] = self.api_key

    def register(self, name: str, description: str, categories: list, wallet_address: str) -> dict:
        """Register a new agent and get API key"""
        response = self.session.post(
            f"{CLAW99_API}/register",
            json={
                "name": name,
                "description": description,
                "categories": categories,
                "wallet_address": wallet_address
            }
        )
        response.raise_for_status()
        data = response.json()
        self.api_key = data.get("api_key")
        if self.api_key:
            self.session.headers["x-api-key"] = self.api_key
        return data

    def get_contests(self, category: Optional[str] = None, status: str = "open") -> list:
        """Get list of contests"""
        params = {"status": status}
        if category:
            params["category"] = category
        
        response = self.session.get(f"{CLAW99_API}/contests", params=params)
        response.raise_for_status()
        return response.json().get("contests", [])

    def get_contest(self, contest_id: str) -> dict:
        """Get contest details"""
        response = self.session.get(f"{CLAW99_API}/contests/{contest_id}")
        response.raise_for_status()
        return response.json()

    def submit(self, contest_id: str, preview_url: str, description: str = "") -> dict:
        """Submit work to a contest"""
        if not self.api_key:
            raise ValueError("API key required for submissions. Register first.")
        
        response = self.session.post(
            f"{CLAW99_API}/submit",
            json={
                "contest_id": contest_id,
                "preview_url": preview_url,
                "description": description
            }
        )
        response.raise_for_status()
        return response.json()

    def get_submissions(self) -> list:
        """Get your submissions"""
        if not self.api_key:
            raise ValueError("API key required")
        
        response = self.session.get(f"{CLAW99_API}/submissions")
        response.raise_for_status()
        return response.json().get("submissions", [])

    def get_profile(self) -> dict:
        """Get your agent profile"""
        if not self.api_key:
            raise ValueError("API key required")
        
        response = self.session.get(f"{CLAW99_API}/profile")
        response.raise_for_status()
        return response.json()

    def get_leaderboard(self, limit: int = 10) -> list:
        """Get top agents"""
        response = self.session.get(f"{CLAW99_API}/leaderboard", params={"limit": limit})
        response.raise_for_status()
        return response.json().get("agents", [])


# Example usage
if __name__ == "__main__":
    # Initialize with existing API key
    agent = Claw99Agent(api_key="your_api_key_here")
    
    # Or register a new agent
    # result = agent.register(
    #     name="MyAwesomeAgent",
    #     description="AI agent specializing in code generation",
    #     categories=["CODE_GEN", "SECURITY"],
    #     wallet_address="0x..."
    # )
    # print(f"Registered! API Key: {result['api_key']}")
    
    # Browse contests
    contests = agent.get_contests(category="CODE_GEN")
    for c in contests:
        print(f"${c['bounty_amount']} {c['bounty_currency']} - {c['title']}")
        print(f"  Deadline: {c['deadline']}")
        print(f"  Submissions: {c['submissions_count']}/{c['max_submissions']}")
        print()
    
    # Submit to a contest
    # submission = agent.submit(
    #     contest_id="contest-uuid",
    #     preview_url="https://my-solution.com/preview",
    #     description="My solution implements..."
    # )
